### DISCUSSION TITLE
Prettier for the template section

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#16672] [General - Components / Directives / etc] [answered]
Hi

I'm using vscode with prettier but it seems that the template part is not pretty as I want it to be. It seems that the indentation and wrapping are done based on line's length or something and not based on html tags and hierarchy.

What are you using to format better the template part? What are the correct settings to make it work based on html tags?
Thanks

**Comment by AmosSzust** [comment]
Any idea what am I missing with the prettier not prettying the template part?
Thanks

**Comment by sullyD64** [comment, marked as answer]
Hi @AmosSzust, I believe you want to use something like [eslint-plugin-vue](https://eslint.vuejs.org/), which can handle both syntactic and stylistic lint rules on Vue SFCs, and its stylistic formatting is more configurable than prettier's basic HTML formatter.

Long time ago I've found myself wanting both formatters (`eslint-plugin-vue` for the `<template>` blocks and `prettier` for the `<script>` blocks and plain js/ts files), but I had to tell eslint to _not_ apply prettier rules to the `<template>`  blocks. To do this, I've included [eslint-plugin-prettier-vue](https://github.com/meteorlxy/eslint-plugin-prettier-vue), which can be configured to disable `prettier` for individual SFC blocks.

Here are some snippets of my setup:

```js
// package.json
{
  ...
  "devDependencies": { 
     "eslint": "^8.55.0",
     "eslint-plugin-prettier-vue": "^5.0.0",
     "eslint-plugin-vue": "^9.19.2",
     "prettier": "^3.0.3",
  }
}
```

```js
// .eslintrc.js
module.exports = {
  extends: {
     // https://github.com/meteorlxy/eslint-plugin-prettier-vue#eslint-config
    "plugin:prettier-vue/recommended",
    // plugin:prettier-vue/recommended includes all functionalities of eslint-plugin-prettier so you do not need it.
    // https://github.com/prettier/eslint-config-prettier#installation
    // "prettier",
    // https://eslint.vuejs.org/rules/#priority-c-recommended-potentially-dangerous-patterns
    "plugin:vue/vue3-recommended",
  },
  ...
  plugins: {
    "vue", // required to lint *.vue SFC files
  },
  settings: {
     "prettier-vue": {
        SFCBlocks: {
          /*
            disables prettier formatting for template blocks in vue single file components,
            so that we can leverage eslint-plugin-vue with the recommended config
            https://github.com/meteorlxy/eslint-plugin-prettier-vue
          */
          template: false,
      },
  }
}
```

```js
// .vscode/settings.json
{
  ...
  // Use prettier via eslint to format typescript, javascript and JSON files
  "[typescript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
  },
  "[javascript]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
  },
  "[json]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
  },
  /*
    For vue single-file-components, we use eslint-plugin-prettier-vue
    (see https://github.com/meteorlxy/eslint-plugin-prettier-vue), which allows us to use:
    - eslint for formatting the <template> section, so that we can leverage eslint-plugin-vue
      (https://eslint.vuejs.org/) with the `recommended` config
    - prettier (via eslint) for formatting the <script> and <style> sections, so that all js/ts code
      has the same code style inside and outside vue SFCs (including CSS). We use stylelint for linting css/scss.
      NB: we DON'T use volar, which relies on vscode's default formatting
  */
  "[vue]": {
    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    // "editor.defaultFormatter": "vue.volar"
  },
}

``` 

Hope anything of this fits your use case :smile:

**Reply by AmosSzust** [reply]
I may say that now it prettier-ed it too much because 

1. It puts each attribute in a different line and not just when the line is too long.
2. I have this additional line of the closing tag that I admit bugs my eye
```
<q-card
          v-for=""
          :key=""
          :class=""
          @click=""
        >
```
I can live with that and get used to it in time but if you have a tip for that, let me know.
Thanks a lot!

**Reply by sullyD64** [reply]
You can try individually disabling or tweaking some rules to your liking, like [max-attributes-per-line](https://eslint.vuejs.org/rules/max-attributes-per-line.html)!
